src/pages/public/ContactForm.js

```diff
@@ -0,0 +1,125 @@
+import React, {useState} from 'react'
+import { useFormStyles} from "../../styles";
+import {Box, Container, Grid, Paper} from "@material-ui/core";
+import Button from '@material-ui/core/Button';
+import CloudUploadIcon from '@material-ui/icons/CloudUpload';
+import {useSelector} from "react-redux";
+import {apiAxios, messagesActions} from "../../redux/actions";
+import Loader from "../../components/common/Loader";
+import FieldGenerator from '../../components/common/FieldGenerator';
+import {useDispatch} from "react-redux";
+
+
+export default function ContactForm() {
+
+    const error = {};
+    const initialStore = {choix: 0, reason: null, message: null, loading: false}
+    const [store, changeStore] = useState(initialStore);
+    const classes = useFormStyles();
+
+    const onChange = event => {
+        const {name, value} = event.target;
+        if (name === 'choix' && choicePossibilites[value - 1]) {
+            changeStore(prevState => ({...prevState, reason: choicePossibilites[value - 1].label}));
+        }
+        changeStore(prevState => ({...prevState, [name]: value}));
+    }
+
+    const Field = new FieldGenerator(
+        store,
+        error,
+        onChange,
+        null,
+        null
+    );
+
+    const choicePossibilites = [
+        {
+            label: 'Création d\'une vente', value: '1'
+        },
+        {
+            label: 'Gestion d\'une vente', value: '2'
+        },
+        {
+            label: 'Commande passée', value: '3'
+        },
+        {
+            label: 'Signaler un bug', value: '4'
+        },
+        {
+            label: 'Autre', value: '5'
+        }
+    ];
+
+    const user = useSelector(store => store.getAuthUser());
+    const dispatch = useDispatch();
+
+    async function handleSubmit() {
+
+        const sender = {name: user.name, id: user.id, email: user.email};
+        const datas = {message: store.message, sender: sender, reason: store.reason};
+        changeStore(prevState => ({...prevState, loading: true}));
+
+        try {
+            await apiAxios.post("/feedback", datas).then(() => {
```

Comment:
```suggestion
            await apiAxios.post("/feedback", datas);
```
No need for `.then()` with `async` functions because `await` will stop and return the response or  raise an error ([check here](https://blog.engineering.publicissapient.fr/2017/11/14/asyncawait-une-meilleure-facon-de-faire-de-lasynchronisme-en-javascript/))